staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -158,3 +159,82 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 
 	return allErrs
 }
+
+// ValidateCustomResourceDefinitionSchema statically validates
+func ValidateCustomResourceDefinitionSchema(JSONSchema *apiextensions.JSONSchemaProps, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if JSONSchema == nil {
+		return allErrs
+	}
+
+	if JSONSchema.UniqueItems == true {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("JSONSchema"), "uniqueItems cannot be set to true since the runtime complexity becomes quadratic"))
```

Comment:
We need a recursive walker for this because it can occur everywhere deep in the schema. Also for those below of course.